### PR TITLE
only conditionally declare distro dependency in setup.py

### LINKED ISSUE #185: On Ubuntu 16.04 Xenial, python-rospkg now depends on non-exsting python-distro
```
$ docker pull ubuntu:16.04
[...]
$ docker run -it ubuntu:16.04
# echo 'deb http://packages.ros.org/ros/ubuntu xenial main'  > /etc/apt/sources.list.d/ros-latest.list
# apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
[...]
# apt-get -y update
[...]
# apt-get -y install python-rospkg-modules
[...]
The following packages have unmet dependencies:
 python-rospkg-modules : Depends: python-distro but it is not installable
E: Unable to correct problems, you have held broken packages.
```

### PR DESCRIPTION
**Pull request by dirk-thomas**
Fixes #185.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,6 +5,14 @@
 
 from setuptools import setup
 
+install_requires = ['catkin_pkg', 'PyYAML']
+
+if (
+    sys.version_info[0] > 3 or
```
Comment (nuclearsandwich) [inline setup.py]: Won't this need import sys?

Comment (dirk-thomas) [reply setup.py]: Line 4?

Comment (nuclearsandwich) [reply setup.py]: My mistake, sys was neither imported nor used in the diff context.

Comment (tfoote) [reply setup.py]: This is a follow up to #182 in that we have the conditional for if it's less that python 3.8. To that end should we have the same check here?

Comment (dirk-thomas) [reply setup.py]: @tfoote I don't understand the question. The condition checks exactly that: if the Python version is 3.8 or higher the `distro` dependency is needed.

Comment (tfoote) [reply setup.py]: sorry I missed the second half under the comment

File: setup.py [hunk 2]
```diff
@@ -5,6 +5,14 @@
...
+    sys.version_info[0] == 3 and sys.version_info[1] >= 8
```
Comment (sloretz) [inline setup.py]: Is the version of python used to call `setup.py` guaranteed to be the same as the version of python on the target system? Does that mean during release the `setup.py` invoked multiple times, once for each target platform?

Comment (dirk-thomas) [reply setup.py]: I agree with your concern. The first commit probably doesn't work.

Please see 9627024db19a3832be7ab2058e7811bbd9bf72e4 for an alternative approach.

**Comment by dirk-thomas** [comment]
The latest state only declares `distro` in the `setup.py` as a dependency when not building the Debian packages. For Debian packages the explicit `Depends3` in the `stdeb.cfg` file should be sufficient. :crossed_fingers:

**Pull request merged**